src/components/GradesView/EditModal/OverrideTable/index.jsx

```diff
@@ -20,18 +20,20 @@ export const OverrideTable = () => {
 
   if (hide) { return null; }
 
+  const tableData = [
+    ...data,
+    {
+      adjustedGrade: <AdjustedGradeInput />,
+      date: formatDateForDisplay(new Date()),
+      reason: <ReasonInput />,
+    },
+  ];
```

Comment:
should we put a check on the data something like 

```suggestion
    const tableData = data ? [
    ...data,
    {
      adjustedGrade: <AdjustedGradeInput />,
      date: formatDateForDisplay(new Date()),
      reason: <ReasonInput />,
    },
  ] : [];
```

While testing I landed up on error when data was undefined and this happened multiple times mostly I guess when the states were getting updated.

Reply:
![Peek 2024-08-02 01-09](https://github.com/user-attachments/assets/70a6c55a-89b3-420e-9410-f176d1ec148b)

Reply:
We should backport https://github.com/openedx/frontend-app-gradebook/commit/4e9270ab8eb6a2e41fa2f4d01e29bfc14a6e165b